### PR TITLE
Update SDK

### PR DESCRIPTION
**Pull request by 333fred**
Move to using .NET 7p5, and language version preview.

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -56,6 +56,10 @@
     <NoWarn>$(NoWarn.Replace('1591', ''))</NoWarn>
     <NoWarn Condition=" '$(BuildingInsideVisualStudio)' == 'true' ">$(NoWarn);0105</NoWarn>
 
+    <!-- Central package management requires that all packages have package source mapping, but we're using all dotnet feeds, and source-build and other downstream rebuilders of our souces
+         would need to support updating that list for their local builds. -->
+    <NoWarn>$(NoWarn);NU1507</NoWarn>
```
Comment (333fred) [inline Directory.Build.props:61]: See https://docs.microsoft.com/en-us/nuget/consume-packages/package-source-mapping for info about mapping. I don't think it matters for our use case: we don't depend on any public feeds, only dnceng feeds, so picking up a maliciously-versioned preview from a public feed like nuget.org isn't an issue.

**Comment by 333fred** [comment]
@dotnet/razor-compiler for reviews.

**Review by chsienki — APPROVED** [review]

File: Directory.Build.props [hunk 2]
```diff
@@ -11,7 +11,7 @@
     <EnableNETAnalyzers>true</EnableNETAnalyzers>
     <AnalysisLevel>latest</AnalysisLevel>
     <AnalysisMode>Default</AnalysisMode>
-    <LangVersion>Latest</LangVersion>
+    <LangVersion>preview</LangVersion>
```
Comment (jcouv) [inline Directory.Build.props:14]: I don't think we should use a preview LangVer in a shipped product, risking depending on a perview feature. 
Can we set this to 11 and upgrade it as needed?

Comment (jcouv) [reply Directory.Build.props:14]: FWIW, it looks like we're using "preview" in roslyn (`settings.props`). Seems strange to me

**Review by jcouv — COMMENTED** [review]
Done with review pass (iteration 2)

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 2)

File: Directory.Build.props (hunk 2 above)
Comment (jcouv) [reply Directory.Build.props:14]: Resolved offline

**Pull request merged**